bugsnag-android-core/src/main/java/com/bugsnag/android/Stacktrace.kt

```diff
@@ -49,34 +73,10 @@ internal class Stacktrace : JsonStream.Streamable {
         }
     }
 
-    private fun limitTraceLength(frames: List<Stackframe>): List<Stackframe> {
+    private fun limitTraceLength(frames: List<Stackframe>): MutableList<Stackframe> {
         return when {
-            frames.size >= STACKTRACE_TRIM_LENGTH -> frames.subList(0, STACKTRACE_TRIM_LENGTH)
-            else -> frames
-        }
-    }
-
-    private fun serializeStackframe(
-        el: StackTraceElement,
-        projectPackages: Collection<String>,
-        logger: Logger
-    ): Stackframe? {
-        try {
-            val className = el.className
-            val methodName = when {
-                className.isNotEmpty() -> className + "." + el.methodName
-                else -> el.methodName
-            }
-
-            return Stackframe(
-                methodName,
-                el.fileName ?: "Unknown",
-                el.lineNumber,
-                inProject(className, projectPackages)
-            )
-        } catch (lineEx: Exception) {
-            logger.w("Failed to serialize stacktrace", lineEx)
-            return null
+            frames.size >= STACKTRACE_TRIM_LENGTH -> frames.subList(0, STACKTRACE_TRIM_LENGTH).toMutableList()
+            else -> frames.toMutableList()
```

Comment:
This could start to generate a lot of heap data... Is it possible to introspect `frames` to see if it's already mutable?